### DISCUSSION TITLE
Serverless Functions How To Access Python Print Statements

### DISCUSSION THREAD

**Discussion by Coderman562** [vercel/community#3125] [Help] [answered]
### Summary

Hello all, how can I access the print statements produced by my serverless function in the logs output? It doesn't seem to produce any when it should. 

I read that there is a console.log but I'm not sure how to have prints go there as I am using python.

### Steps to Reproduce

Add a print statement to a to a serverless function try printing it out when its called

**Comment by orphiano-0** [comment]
I suggest using 'logging' module instead of 'print' function. 

```
import logging

def function(event, context):
    # Configure the logging
    logging.basicConfig(level=logging.INFO)

    # Log statement
    logging.info("This is a log message.")

    # Your code...

    return 
```

**Reply by voynow** [reply]
Using logging and still not seeing logs produced in the logs tab:

```
logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

@app.post("/extract_metrics")
async def extract_metrics(repo: Dict[str, str]):
        logger.info("Inside /extract_metrics endpoint")
        for filepath, code in repo.items():
                logger.info(f"Composing metrics for {filepath}")
                composite_metrics[filepath] = run_some_function()
```

No logs produced - any other serverless function/python users experience this on Vercel?

**Reply by voynow** [reply, marked as answer]
So apparently print statements work better..... https://github.com/orgs/vercel/discussions/3197

**Reply by vasanth-asokan** [reply]
I am using both prints() and logger() (technically to vercel this shouldn't make any difference), and I see only `print` showing up in my runtime logs. I am also emitting logs to logtail via directly using their python package (it goes through Python's `logging` and that also seems to get disabled in Vercel production, works fine in local)

@amyegan Is the Vercel runtime doing something to disable output from the python logging module in it's production serverless runtimes?
